### PR TITLE
Update s3 cache to support folder

### LINKED ISSUE #1493: [Feature]: Add ability to cache to specific folder in S3
### The Feature

I would like to be able to cache my requests in a specific folder in s3. 

e.g., 

```python
# pass s3-bucket name
litellm.cache = Cache(type="s3", s3_bucket_name="cache-bucket-litellm", s3_region_name="us-west-2", folder="litellm-cache")


```

### Motivation, pitch

I have other things in my bucket and want to keep it clean... 

### Twitter / LinkedIn details

https://www.linkedin.com/in/duarteocarmo/

### PR DESCRIPTION
**Pull request by duarteocarmo**
resolves #1493

### CODE REVIEW

File: litellm/integrations/s3.py [hunk 1]
```diff
@@ -122,8 +124,9 @@ def log_event(self, kwargs, response_obj, start_time, end_time, print_verbose):
                     pass
 
             s3_object_key = (
+                (self.s3_path.rstrip('/') + '/' if self.s3_path else '') +
```
Comment (ishaan-jaff) [inline litellm/integrations/s3.py]: any reson why s3 path should be part of the key ?

File: litellm/integrations/s3.py [hunk 2]
```diff
@@ -122,8 +124,9 @@ def log_event(self, kwargs, response_obj, start_time, end_time, print_verbose):
                     pass
 
             s3_object_key = (
+                (self.s3_path.rstrip('/') + '/' if self.s3_path else '') + 
                 payload["id"] + "-time=" + str(start_time)
-            )  # we need the s3 key to include the time, so we log cache hits too
```
Comment (ishaan-jaff) [inline litellm/integrations/s3.py:126]: can we not delete this comment😅

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]
requested changes, can you add this param to docs too

File: litellm/integrations/s3.py (hunk 2 above)
Comment (duarteocarmo) [reply litellm/integrations/s3.py:126]: Reverted, sorry!

File: litellm/integrations/s3.py (hunk 1 above)
Comment (duarteocarmo) [reply litellm/integrations/s3.py]: I believe that is the way you point to a folder in boto3. 

Could not find better docs then here.. 

https://stackoverflow.com/questions/39272397/uploading-file-to-specific-folder-in-s3-using-boto3#comment65879979_39272397

**Comment by duarteocarmo** [comment]
Let me know if this works :)

**Comment by duarteocarmo** [comment]
Also updated docs!

**Review by ishaan-jaff — APPROVED** [review]
lgtm ! thanks for the contribution !

**Review by ishaan-jaff — APPROVED** [review]

**Pull request merged**